server/test/categoryRouter.test.js

```diff
@@ -0,0 +1,74 @@
+const Category = require('../db/models/category');
+const server = require('../index');
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const { getMongoDB } = require('../db');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const mongoose = require('mongoose');
+const User = require('../db/models/user');
+
+let exampleCategory;
+let defaultUser;
+let mongoServer;
+const expect = chai.expect;
+chai.use(chaiHttp);
+
+function handleErr(err) {
+  expect(err).to.be.null;
+}
+
+before(async function () {
+  mongoServer = new MongoMemoryServer();
+  const mongoUri = await mongoServer.getUri();
+  getMongoDB(mongoUri);
+  defaultUser = new User({ email: 'lobster-ice-cream-lover@gmeil.com' });
+  defaultUser.save(handleErr);
+  exampleCategory = new Category({ title: 'hello there', userId: defaultUser._id });
+  exampleCategory.save(handleErr);
+});
+
+after(async function () {
+  await mongoose.disconnect();
+  await mongoServer.stop();
+});
+
+it('There should be one user in the database', async function () {
+  await Category.find(function (err, categories) {
+    expect(err).to.be.null;
+    console.log(categories);
+  });
```

Comment:
Is this a mix of the await and callback style? Please move the expect statements below the await call.

Reply:
Sorry about that! I didn't have a good understanding of the two styles